Libraries/oneVPL/dpcpp-blur/src/dpcpp-blur.cpp

```diff
@@ -184,209 +187,149 @@ int main(int argc, char **argv) {
                                    cliParams.implValue);
   VERIFY(MFX_ERR_NONE == sts, "MFXSetConfigFilterProperty failed for Impl");
 
-  sts = MFXCreateSession(loader, 0, &session);
+  // Match oneVPL implementation with sycl::q device
+  while (MFX_ERR_NOT_FOUND != MFXEnumImplementations(loader, sessionIdx,
+                  MFX_IMPLCAPS_IMPLDESCSTRUCTURE, (mfxHDL*)&implDesc)) {
+    mfxU32 devID = (mfxU32)std::stoi(implDesc->Dev.DeviceID, nullptr, 16);
+    if (devID == desiredDevice) {
+      desiredSessionIdx = (int)sessionIdx;
+      break;
+    }
+    MFXDispReleaseImplDescription(loader, implDesc);
+    sessionIdx++;
+  }
+
+  VERIFY(sessionIdx >= 0,
```

Comment:
This should probably be desiredSessionIdx.

Reply:
Then, 
`sts = MFXCreateSession(loader, sessionIdx, &session);`
=>
`sts = MFXCreateSession(loader, desiredSessionIdx, &session);`
right?